core/integration/client_tests.go

```diff
@@ -166,10 +171,48 @@ func TestBatchUpdate(ctx context.Context, env *Env, t *testing.T) []*tpb.Action
 	return nil
 }
 
+func NewClientWithTracker(t *testing.T, env *Env) (*client.Client, *tracker.LogTracker) {
+	t.Helper()
+
+	config := env.Directory
+	logVerifier, err := tclient.NewLogVerifierFromTree(config.GetLog())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mapVerifier, err := tclient.NewMapVerifierFromTree(config.GetMap())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// VRF key
+	vrfPubKey, err := p256.NewVRFVerifierFromRawKey(config.GetVrf().GetDer())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	logTracker := tracker.New(logVerifier)
+
+	verifier := verifier.New(vrfPubKey, mapVerifier, logVerifier, logTracker)
+
+	minInterval, err := ptypes.Duration(config.MinInterval)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cli := client.New(env.Cli, config.DirectoryId, minInterval, verifier)
+	return cli, logTracker
+}
+
 // TestEmptyGetAndUpdate verifies set/get semantics.
 func TestEmptyGetAndUpdate(ctx context.Context, env *Env, t *testing.T) []*tpb.Action {
 	go runSequencer(ctx, t, env.Directory.DirectoryId, env)
 
+	cli, logTracker := NewClientWithTracker(t, env)
+	logTracker.SetUpdatePredicate(func(_, newRoot types.LogRootV1) bool {
+		return newRoot.TreeSize%5 == 1
```

Comment:
Why do we need to test this predicate, and not the original one?

Reply:
Good question.
a) The original predicate is tested in the tracker's unit tests. 
b) The purpose of this is to produce more interesting consistency proofs in the test vectors.  If we updated the log root every time, the consistency proofs would always be empty.